markers/sauce.py

```diff
@@ -14,7 +16,27 @@ def pytest_addoption(parser):
 def pytest_configure(config):
     config.addinivalue_line('markers', __doc__.splitlines()[0])
     if config.option.sauce:
+        conf.env['browser']['sauce'] = True
         if config.option.markexpr:
             config.option.markexpr = 'sauce and ({})'.format(config.option.markexpr)
         else:
             config.option.markexpr = 'sauce'
+
+
+def pytest_runtest_setup(item):
+    if item.config.option.sauce:
+        conf.env['browser']['sauce'] = True
+        if 'chrome' in str(item).lower():
+            browser_name = 'chrome'
+        elif 'firefox' in str(item).lower():
+            browser_name = 'firefox'
+        else:
+            browser_name = 'firefox'
+        conf.env['browser']['browserName'] = browser_name
+        conf.env['browser']['itemName'] = item.name
+        browser.ensure_browser_open()
+
+
+def pytest_runtest_teardown(item):
+    if item.config.option.sauce:
```

Comment:
@psav: added check "item.config.option.sauce" before teardown, so that only browser closed per test only for --sauce